app/scripts/TrackRenderer.jsx

```diff
@@ -93,29 +312,52 @@ class TrackRenderer extends React.Component {
    * Zooming changes the domain of the scales.
    *
    * Resizing changes the range. Both trigger a rerender.
+   *
+   * @param {TrackRendererProps} props
    */
   constructor(props) {
     super(props);
+    /** @type {boolean} */
     this.dragging = false; // is this element being dragged?
+    /** @type {WithZoomTransform<HTMLElement> | null} */
     this.element = null;
+    /** @type {HTMLElement | null} */
+    this.eventTracker = null;
+    /** @type {HTMLElement | null} */
+    this.eventTrackerOld = null;
+    /** @type {boolean} */
     this.closing = false;
 
+    /** @type {number} */
     this.yPositionOffset = 0;
+    /** @type {number} */
     this.xPositionOffset = 0;
+    /** @type {number} */
     this.scrollTop = 0;
 
+    /** @type {ReturnType<typeof setTimeout> | null} */
     this.scrollTimeout = null;
+    /** @type {number} */
     this.activeTransitions = 0;
 
+    /** @type {import('d3-zoom').ZoomTransform} */
     this.zoomTransform = zoomIdentity;
+    /** @type {() => void} */
     this.windowScrolledBound = this.windowScrolled.bind(this);
+    /** @type {(event?: import('d3-zoom').D3ZoomEvent<HTMLElement, unknown>) => void} */
     this.zoomStartedBound = this.zoomStarted.bind(this);
+    /** @type {(event: import('d3-zoom').D3ZoomEvent<HTMLElement, unknown> & { shiftKey?: boolean }) => void} */
     this.zoomedBound = this.zoomed.bind(this);
+    /** @type {() => void} */
     this.zoomEndedBound = this.zoomEnded.bind(this);
 
+    /** @type {string} */
     this.uid = slugid.nice();
+
+    /** @type {string} */
     this.viewUid = this.props.uid;
 
+    /** @type {unknown} */
```

Comment:
For those familiar with implementing plugin tracks, this is `HGC`. We should try to strictly type this (or make is generic) for eventual TrackRenderer.

Reply:
Agreed strictly typing this would be so helpful for plugin tracks! ([Link](https://github.com/gosling-lang/gosling.js/blob/f17e69f8ea9d83e9c574e864b7dfca94a49fa328/src/missing-types.d.ts#L6-L11) to the HGC type in Gosling for reference)